src/commands/register.ts

```diff
@@ -2,142 +2,161 @@ import { flags } from '@oclif/command';
-          cli.action.stop(chalk.green(`Successfully pushed Docker image for ${image}`));
-        }
+    for (const config_path of config_paths) {
+      await this.register_component(config_path, flags.tag);
+    }
+  }
 
-        const [account_name, _] = node.service_config.getName().split('/');
-        const selected_account = this.accounts.rows.find((a: any) => a.name === account_name);
-        if (!selected_account) {
-          throw new Error(`You do not have access to the account specified in your service config: ${account_name}`);
-        }
+  private async register_component(config_path: string, tag: string) {
+    const component_config = await ComponentConfigBuilder.buildFromPath(config_path);
 
-        cli.action.start(chalk.blue(`Running \`docker inspect\` on the given image: ${image}`));
-        const digest = await getDigest(image).catch(e => {
-          cli.action.stop(chalk.red(`Inspect failed`));
-          throw new Error(`The image specified in your ServiceConfig is not reachable by docker: ${node.service_config.getImage()}`);
-        });
-        cli.action.stop(chalk.green(`Image verified`));
-
-        const image_without_tag = strip_tag_from_image(image); // we don't need the tag on our image because we use the digest as the key.
-
-        const service_dto = {
-          tag: flags.tag,
-          digest: digest,
-          image: image_without_tag,
-          config: classToPlain(node.service_config),
-        };
-        cli.action.start(chalk.blue(`Registering service ${node.service_config.getName()}:${flags.tag} with Architect Cloud...`));
-        await this.post_service_to_api(service_dto, selected_account.id);
-        cli.action.stop(chalk.green(`Successfully registered service`));
-      }
+    const [account_name, _] = component_config.getRef().split('/');
+    const selected_account = this.accounts.rows.find((a: any) => a.name === account_name);
+    if (!selected_account) {
+      throw new Error(`You do not have access to the account specified in your component config: ${account_name}`);
+    }
+
+    for (const [service_ref, service_config] of Object.entries(component_config.getServices())) {
+      const image_tag = `${this.app.config.registry_host}/${component_config.getName()}-${service_config.getName()}:${tag}`;
+      const image = await this.build_docker_image_if_necessary(config_path, service_config, image_tag);
+      component_config.getServiceByRef(service_ref)?.setImage(image);
+    }
+
+    const component_dto = {
+      tag: tag,
+      config: classToPlain(component_config),
+    };
+
+    cli.action.start(chalk.blue(`Registering component ${component_config.getName()}:${tag} with Architect Cloud...`));
+    await this.post_component_to_api(component_dto, selected_account.id);
+    cli.action.stop(chalk.green(`Successfully registered component`));
+  }
+
+
+  private async build_docker_image_if_necessary(config_path: string, service_config: ServiceConfig, image_tag: string) {
```

Comment:
Should this be named `push_image_if_necessary` or something? I feel like the pushing here is the more important part